fix: add identify metrics call at metrics instance setup

NicolasMassart opened the pull request:
## **Description**

adds identify call at the end of metrics config

## **Related issues**

Fixes #8885

## **Manual testing steps**

```gherkin
Feature: identify on start

  Scenario: fresh install
    Given app is a fresh installed
    And first screen appears
    Then logs should display "MetaMetrics configured with ID: [UUID]"
    And no event is sent

  Scenario: wallet created
    Given wallet was created
    And user accepted metrics
    And first screen appears
    Then logs should display "INFO  IDENTIFY event saved {...}"
    And at least 1 event is sent
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

No IDENTIFY log at startup

### **After**

<img width="500" alt="image" src="https://github.com/MetaMask/metamask-mobile/assets/4677568/2042f2fa-62a9-4bed-be61-3654ea16f581">

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

NicolasMassart committed:
add identify metrics call at metrics instance setup

fixes #8885

sethkfman reviewed (dismissed):
LGTM

NicolasMassart committed:
Merge branch 'main' into fix/8885_add_identify

Cal-L commented:
@NicolasMassart Created a fix for the unit test pointing to this PR - https://github.com/MetaMask/metamask-mobile/pull/8928

Cal-L committed:
fix: Fix unit test failure for trackDappViewedEvent (#8928)

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR fixes the test failure for
`app/util/metrics/trackDappViewedEvent/index.test.ts`. It does so by
mocking the module
`../../../util/metrics/UserSettingsAnalyticsMetaData/generateUserProfileAnalyticsMetaData`,
which was causing the failure.

## **Related issues**

Fixes:

## **Manual testing steps**

Run `yarn jest app/util/metrics/trackDappViewedEvent/index.test.ts` and
notice that the test passes

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->
<img width="1176" alt="image"
src="https://github.com/MetaMask/metamask-mobile/assets/10508597/72f97136-0bba-46f6-a88c-df96854bc105">


## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it
is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
- [ ] In case it's "ready for review", I've changed it from "draft" to
"non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

Cal-L committed:
Merge branch 'main' into fix/8885_add_identify

Cal-L reviewed (approved):
LGTM
